de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/wizards/drop_files/DropFilesDialog.java

```diff
@@ -1,39 +1,65 @@
-package de.tu_bs.cs.isf.e4cf.parts.project_explorer.wizards;
+package de.tu_bs.cs.isf.e4cf.parts.project_explorer.wizards.drop_files;
 
 import java.io.IOException;
 import java.nio.file.FileAlreadyExistsException;
 import java.nio.file.Files;
 import java.nio.file.Path;
 
 import org.eclipse.e4.core.contexts.IEclipseContext;
-import org.eclipse.jface.wizard.Wizard;
 
+import de.tu_bs.cs.isf.e4cf.core.stringtable.E4CFileTable;
 import de.tu_bs.cs.isf.e4cf.core.util.RCPMessageProvider;
 import de.tu_bs.cs.isf.e4cf.core.util.services.RCPImageService;
 import de.tu_bs.cs.isf.e4cf.parts.project_explorer.DropElement;
+import javafx.scene.control.Alert;
+import javafx.scene.control.ButtonType;
+import javafx.scene.control.Alert.AlertType;
+import javafx.stage.Stage;
+import javafx.scene.control.DialogPane;
 
 /**
  * A wizard that handles drops from the system explorer to the project explorer,
  * if the drop contains a nested element (e.g. directory)
  */
-public class DropWizard extends Wizard {
+public class DropFilesDialog {
 
-	private ImportDirectoryPage copyOptionsPage;
 	private DropElement dropElement;
-
+	private Alert alert;
+	private ImportDirectoryPage copyOptionsPage;
 	// indicates whether a file has been moved.
 	boolean didFileMove = true;
 
-	public DropWizard(IEclipseContext context, DropElement dropElement, RCPImageService imgService) {
+	/**
+	 * @param context
+	 * @param dropElement
+	 * @param imgService
+	 */
+	public DropFilesDialog(IEclipseContext context, DropElement dropElement, RCPImageService imgService) {
 		this.dropElement = dropElement;
-		this.copyOptionsPage = new ImportDirectoryPage("Import directory", dropElement.getTarget(), context,
-				imgService.getImageDescriptor(null, "icons/Explorer_View/items/folder24.png"));
+		
+		alert = new Alert(AlertType.NONE);
+		alert.setTitle("Import a Directory");
+		
+		final Stage stage = (Stage) alert.getDialogPane().getScene().getWindow();
+		stage.getIcons().add(imgService.getFXImage(null, E4CFileTable.FRAMEWORK_LOGO_SMALL).getImage());
+
+		final DialogPane pane = alert.getDialogPane();
+		pane.getButtonTypes().addAll(ButtonType.OK, ButtonType.CANCEL);

```

Comment:
Cancel Button does not cancel the operation

Reply:
I did not change this behaviour, it just cancels the dialog and takes the default value of copy strategy EMPTY. Should i remove the button, change it's name, or... ?

Reply:
either remove the button or add the possibility of cancelling the drop

Reply:
canceling will no longer default the behaviour to Empty, instead cancel the folder drop (it will still move  files regardless of choice)